salt/modules/linux_sysctl.py

```diff
@@ -231,7 +234,7 @@ def persist(name, value, config=None):
         nlines.append('{0} = {1}\n'.format(name, value))
     try:
         with salt.utils.files.fopen(config, 'w+') as _fh:
-            _fh.writelines(nlines)
+            _fh.writelines(salt.utils.stringutils.to_str(nlines))
```

Comment:
nlines is a list here, so this will raise a TypeError. This could be fixed using:

```python
with salt.utils.files.fopen(config, 'wb') as _fh:
    _fh.writelines(salt.utils.data.encode(nlines))
```

Reply:
Note: this opens the filehandle for binary so that it works with `salt.utils.data.encode()`, which will produce a list of bytestrings.